Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/PlayerInputManager.cs

```diff
@@ -661,6 +664,31 @@ private bool IsDeviceUsableWithPlayerActions(InputDevice device)
             return false;
         }
 
+        private void ValidateInputActionAsset()
+        {
+            if (m_PlayerPrefab == null || m_PlayerPrefab.GetComponentInChildren<PlayerInput>() == null)
+                return;
+
+            var actions = m_PlayerPrefab.GetComponentInChildren<PlayerInput>().actions;
+            if (actions == null)
+                return;
+
+            var isValid = true;
+            foreach (var controlScheme in actions.controlSchemes)
+            {
+                if (controlScheme.deviceRequirements.Count > 0)
+                    break;
+
+                isValid = false;
+            }
+
+            if (!isValid)
+                Debug.LogWarning("The input action asset in the player prefab assigned to PlayerInputManager has " +
+                    "no control schemes with required devices. The JoinPlayersWhenButtonIsPressed join behavior " +
+                    "will not work unless the expected input devices are listed as requirements in the input " +
+                    "action asset.");
```

Comment:
Makes it hard to track down the source. Probably good to mention asset name (or even path?). Also should give player prefab to `LogWarning` call as Unity object argument so that when clicking on the message in the console, the user can jump to the prefab.